### DISCUSSION TITLE
设置退出最大化时候的窗口大小

### DISCUSSION THREAD

**Discussion by flowerseacar** [hooke007/mpv_PlayKit#460] [mpv 通用]
linux下，自定义退出最大化时候的窗口尺寸，和原来视频的分辨率一样。大概就是alt+1这个快捷键实现的效果。
主要问题就是linux下似乎无法自定义退出最大化时候的窗口尺寸

**Comment by hooke007** [comment]
最小化是指窗口缩到任务栏或后台，这个时候连前台窗口都没了还哪来的尺寸？

**Comment by flowerseacar** [comment]
> 最小化是指窗口缩到任务栏或后台，这个时候连前台窗口都没了还哪来的尺寸？

sorry,不是最小化，最大化那个按键在按一次退出最大化的那个时候

**Reply by hooke007** [reply]
> 最大化那个按键在按一次退出最大化的那个时候

这个东西不就是 ？
https://github.com/hooke007/MPV_lazy/blob/2027fb8b2ec766896773c6803c9b7a33a4fc6f12/portable_config/mpv.conf#L82

**Comment by flowerseacar** [comment]
> 最小化是指窗口缩到任务栏或后台，这个时候连前台窗口都没了还哪来的尺寸？

总之就跟windows的potplayer一样就可以了，有木有办法实现和potplayer一样的窗口非最大化尺寸。用potplayer太方便了，可是linux没有

**Reply by hooke007** [reply]
我不用pot，建议上视频或者图

**Comment by flowerseacar** [comment]
> 我不用pot，建议上视频或者图

我说的更简单点，就是你视频最大化的时候，你退出最大化之后在按alt+1,我要的就是退出最大化之后他自动alt+1.就是这个效果。这样他既尺寸和原来不变，我又可以拖动它，甚至拖动到电脑的屏幕外

**Comment by flowerseacar** [comment]
> > 最大化那个按键在按一次退出最大化的那个时候
> 
> 这个东西不就是 ？
> 
> https://github.com/hooke007/MPV_lazy/blob/2027fb8b2ec766896773c6803c9b7a33a4fc6f12/portable_config/mpv.conf#L82

我刚刚又回复了一个，更清晰点，autofit-smaller 这个可以保证他是视频原来的尺寸吗

**Comment by flowerseacar** [comment]
> > 最大化那个按键在按一次退出最大化的那个时候
> 
> 这个东西不就是 ？
> 
> https://github.com/hooke007/MPV_lazy/blob/2027fb8b2ec766896773c6803c9b7a33a4fc6f12/portable_config/mpv.conf#L82

autofit-smaller=100%x100%不起作用，他在退出最大化时候的窗口尺寸无法实现和原来一样啊

**Reply by hooke007** [reply]
> 我刚刚又回复了一个，更清晰点，autofit-smaller 这个可以保证他是视频原来的尺寸吗

建议同个话题在同一楼下发，不然你一句话一个楼我也很混乱你的前后逻辑。

> 你退出最大化之后在按alt+1

mpv的默认alt+1是强制缩放到一倍尺寸，你对这个快捷键的预期是错误的

> autofit-smaller=100%x100%不起作用

我只在windows11测试过它的效果（指点窗口最大化的按钮）

**Reply by flowerseacar** [reply]
不是错误预期，就是原来的尺寸也可以，至少能保证原像素，退而求其次也可以了，因为我同时设置了autofit-larger=1920x1080。大概就是退出最大化的时候的尺寸有没有办法自定义。你在windows上测试的autofit-smaller=100%x100%是起作用的吗。

**Reply by hooke007** [reply]
你要的到底是退出最大化时窗口尺寸=视频分辨率
还是
退出最大化时窗口尺寸=mpv启动窗口的尺寸？

**Reply by flowerseacar** [reply]
实在不行的话，我只能退出最大化的时候手动按alt+1实现了。我的是linux系统，我系统上autofit-smaller=100%x100%不起作用，他退出最大化的时候尺寸似乎是大概原来屏幕的百分之65左右吧

**Reply by hooke007** [reply]
> 我只能退出最大化的时候手动按alt+1

那你为什么不把这两个命令绑在一个按键上

**Reply by flowerseacar** [reply]
退出最大化时窗口尺寸=视频分辨率 这个是我想要的

**Reply by flowerseacar** [reply]
对，是可以。但我不会啊，而且我是刚刚才发现alt+1有这种功能。怎么实现啊

**Reply by flowerseacar** [reply]
而且我不仅仅是想要按键绑定实现，而且我还想要他手动点击退出最大化的按钮的时候，他也能实现我的这种效果

**Reply by hooke007** [reply]
"退出最大化时窗口尺寸"
你为什么不直接alt+1而是先退最大化再alt+1。。。

**Reply by flowerseacar** [reply]
set window-maximized no ; set window-scale 1 这2个命令不行啊，哥们
为啥不直接alt+1.是因为视频分辨率大于等于我的显示器分辨率，打开的时候他就直接最大化了。而且我也想保持这个功能。
总之，我就是想退出最大化的时候的尺寸是可以自定义的，然后我把他自定成视频的分辨率

**Reply by flowerseacar** [reply]
我发现最大化的时候直接按alt+1一样可以实现退出最大化的时候的取消固定，并且可以拖动，我去。

**Reply by flowerseacar** [reply]
行吧，这问题就这么解决。但是我确实想要通过图形界面的点击也可以实现这个效果，而不仅仅是按键。
同时linux下真没法自定义退出最大化时候的窗口尺寸

**Reply by hooke007** [reply]
> 视频分辨率大于等于我的显示器分辨率

windows上无论视频什么分辨率都不会超出屏幕
此外mpv的其他窗口行为也会受不同桌面环境的限制和改变
这也是为什么我这边行的方案你那边都不行

已添加linux标签，另外建议在主题下第一楼应该尽可能详尽表述清楚，这样可以更快定位真实问题和解决方案。

**Reply by flowerseacar** [reply]
等会我修改下1楼
